project_template/_base/project_name/apps/common/mixins/admin.py

```diff
@@ -0,0 +1,10 @@
+class ReadOnlyAdminMixin:  # pragma: no cover
+
```

Comment:
`Explicit is better than implicit.` ! 

In Django 2.1 `has_view_permission` is added!

@alexryabtsev what do you think? Maybe should add the method as well (will return True) ?

Reply:
We don't need to add `has_view_permission`. In this case we need to rely on Django implementation.

By default if user has permissions for table he able to do all CRUD operations. The idea of `ReadOnlyAdminMixin` is to deny create, update, delete operations even if they are available by users permissions. It could be a case when this operations are denied by design (some settings, currency list, etc) and editable only via data migrations.

Reply:
> In this case we need to rely on Django implementation.

In that case make sense. Btw nice mixin

Reply:
Looks like good snippet, but I'm not sure that it's reasonable to put it to skeleton.

Reply:
I've got this in my three recent project, so I find it very useful. It's quite simple but does its job and I don't need to write it again or copy/paste each time.

The same for Redis client, Enumeration and all the stuff.

My point is that skeleton should reduce the time to start project. I want to start project and and have all the stuff in place from the beginning.

Is there a project where we haven't got Redis client, enumeration or simple helper for uploading media files? I don't think so.